seqr/views/utils/terra_api_utils.py

```diff
@@ -67,6 +69,12 @@ def anvil_call(method, path, access_token, user=None, headers=None, root_url=Non
     url, headers = _get_call_args(path, headers, root_url)
     request_func = getattr(requests, method)
     headers.update({'Authorization': 'Bearer {}'.format(access_token)})
+    cache_key = 'terra_req__' + hashlib.md5((url + str(headers) + str(kwargs)).encode('utf-8')).hexdigest()
```

Comment:
This is not a user specific cache, which means this will not do what we need it to. For example, if you list workspaces a user has access to, and then 3 seconds later a different user makes that call, they will get back the previous users access list which is really bad!

I also don't think we should be caching all anvil requests, just the ones that get done really frequently as part of authorization. I would move this caching up a level into the function we actually want cached, and make sure the username is in the cache key

Reply:
Also worth noting that codacy is flagging this line for "Use of insecure MD2, MD4, MD5, or SHA1 hash function."
By moving it up a level so you know what is being called, you can make a simpler cache key without needing to encode the whole request. For example, in `user_get_workspace_access_level` you could make the cache key something like `'terra_req__{}__{}'.format(path, user.username)`

Reply:
I thought it has no harm to allow other users to get permission information for a project. But, I know it is wrong at least the workspace list must be per-user based.

Reply:
I did a test for different users to send the same request but they don't hit the same cache for they have different headers which include the bearer tokens. But your suggestion is better because it is neater.